src/main/java/it/gov/pagopa/pu/fileshare/service/export/ExportFileFacadeServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package it.gov.pagopa.pu.fileshare.service.export;
+
+
+import it.gov.pagopa.pu.fileshare.connector.processexecutions.ExportFileService;
+import it.gov.pagopa.pu.fileshare.dto.FileResourceDTO;
+import it.gov.pagopa.pu.fileshare.exception.custom.UnauthorizedFileDownloadException;
+import it.gov.pagopa.pu.fileshare.service.AuthorizationService;
+import it.gov.pagopa.pu.fileshare.service.FileStorerService;
+import it.gov.pagopa.pu.fileshare.service.UserAuthorizationService;
+import it.gov.pagopa.pu.p4paauth.dto.generated.UserInfo;
+import it.gov.pagopa.pu.p4paprocessexecutions.dto.generated.ExportFile;
+import java.io.InputStream;
+import java.nio.file.Path;
+import org.springframework.core.io.InputStreamResource;
+
+public class ExportFileFacadeServiceImpl implements ExportFileFacadeService {
+  private final UserAuthorizationService userAuthorizationService;
+  private final FileStorerService fileStorerService;
+  private final ExportFileService exportFileService;
+
+  public ExportFileFacadeServiceImpl(
+    UserAuthorizationService userAuthorizationService,
+    FileStorerService fileStorerService,
+    ExportFileService exportFileService) {
+    this.userAuthorizationService = userAuthorizationService;
+    this.fileStorerService = fileStorerService;
+    this.exportFileService = exportFileService;
+  }
+
+  @Override
+  public FileResourceDTO downloadExportFile(Long organizationId,
+    Long exportFileId, UserInfo user,
+    String accessToken) {
+    userAuthorizationService.checkUserAuthorization(organizationId, user,
+      accessToken);
+
+    ExportFile exportFile = exportFileService.getExportFile(exportFileId, accessToken);
+
+    if (!AuthorizationService.isAdminRole(organizationId, user) &&
```

Comment:
If exportFile is null you should throw notFoundException (is this behavior not implemented for ingestionFlowFile? If not, please, let's fix it

If you add a test on with exportFile null you wil see that actually you will have a NullpointerException